kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -482,12 +490,38 @@ private Optional<CRCInfo> buildPostCommitCrcInfoIfCurrentCrcAvailable(
   /**
    * Get the part of the schema of the table that needs the statistics to be collected per file.
    *
-   * @param engine {@link Engine} instance to use.
-   * @param transactionState State of the transaction
+   * @param transactionState State of the transaction.
    * @return
    */
-  public static List<Column> getStatisticsColumns(Engine engine, Row transactionState) {
-    // TODO: implement this once we start supporting collecting stats
-    return Collections.emptyList();
+  public static List<Column> getStatisticsColumns(Row transactionState) {
+    int numIndexedCols =
+        TableConfig.DATA_SKIPPING_NUM_INDEXED_COLS.fromMetadata(
+            TransactionStateRow.getConfiguration(transactionState));
+
+    // Get the list of partition columns to exclude
+    Set<String> partitionColumns =
+        TransactionStateRow.getPartitionColumnsList(transactionState).stream()
```

Comment:
nit: can we just do `new HashSet<>(TransactionStateRow.getPartitionColumnsList(transactionState))` ? it avoids the overhead of all the stream creation